distributed/client.py

```diff
@@ -2656,9 +2668,8 @@ def __aiter__(self):
         return self
 
     def __next__(self):
-        with self.lock:
-            if not self.futures and self.queue.empty():
-                raise StopIteration()
+        if self.is_empty():
```

Comment:
Yeah, the async code doesn't need locks as long as we're consistent between yield/await points.